setup/action.yml

```diff
@@ -0,0 +1,56 @@
+name: Setup
+description: "Set up the Release Environment"
+inputs:
+  aws_role_arn:
+    description: "The aws role to acquire"
+    required: true
+  aws_region_name:
+    description: "The aws region to use"
+    required: true
+  aws_secret_id:
+    description: "The name of the aws secret to use"
+    required: true
+  app_id:
+    description: "The GitHub App id"
+    required: true
+  private_key:
+    description: "The GitHub App private key"
+    required: true
+  artifactory_registry:
+    description: "Artifactory registry to be used"
+    default: artifactory.corp.mongodb.com
+  artifactory_image:
+    description: "Image to use for artifactory"
+    default: release-tools-container-registry-local/garasign-git
+
+runs:
+  using: composite
+  steps:
+  - uses: actions/create-github-app-token@v1
+    id: app-token
+    with:
+      app-id: ${{ inputs.app_id }}
+      private-key: ${{ inputs.private_key }}
+  - uses: actions/checkout@v4
+    with:
+      token: ${{ steps.app-token.outputs.token }}
```

Comment:
For PHP, we have some workflows that need to use `fetch-depth: 0`, while others need to use `submodules: true`. Since it's not possible to set these, the setup action is only half useful for us.

Is the checkout strictly required here or is this more of a convenience? If it's the latter, I don't mind removing this and running checkout in the workflow that leverages this action; if it's strictly required I'd kindly ask to add inputs for `fetch-depth` and `submodules` and pass those on to the checkout action.

Reply:
I'll move that part out of the action.